simulation/traffic_simulator/src/hdmap_utils/hdmap_utils.cpp

```diff
@@ -85,6 +85,50 @@ HdMapUtils::HdMapUtils(
     lanelet::utils::query::shoulderLanelets(lanelet::utils::query::laneletLayer(lanelet_map_ptr_));
 }
 
+auto HdMapUtils::gelAllCanonicalizedLaneletPoses(
+  const traffic_simulator_msgs::msg::LaneletPose & lanelet_pose) const
+  -> std::vector<traffic_simulator_msgs::msg::LaneletPose>
+{
+  std::vector<traffic_simulator_msgs::msg::LaneletPose> canonicalized_all;
+  if (lanelet_pose.s < 0) {
+    if (const auto ids = getPreviousLaneletIds(lanelet_pose.lanelet_id); ids.empty()) {
+      return canonicalized_all;
+    } else {
+      for (const auto id : ids) {
+        const auto lanelet_pose_tmp = traffic_simulator::helper::constructLaneletPose(
+          id, lanelet_pose.s + getLaneletLength(id), 0);
+        if (auto canonicalized_lanelet_poses = gelAllCanonicalizedLaneletPoses(lanelet_pose_tmp);
+            canonicalized_lanelet_poses.empty()) {
+          canonicalized_all.push_back(lanelet_pose_tmp);
+        } else {
+          std::copy(
+            canonicalized_lanelet_poses.begin(), canonicalized_lanelet_poses.end(),
+            std::back_inserter(canonicalized_all));
+        }
+      }
+    }
+  }
+  if (lanelet_pose.s > (getLaneletLength(lanelet_pose.lanelet_id))) {
+    if (const auto ids = getNextLaneletIds(lanelet_pose.lanelet_id); ids.empty()) {
+      return canonicalized_all;
+    } else {
+      for (const auto id : ids) {
+        const auto lanelet_pose_tmp = traffic_simulator::helper::constructLaneletPose(
+          id, lanelet_pose.s - getLaneletLength(lanelet_pose.lanelet_id), 0);
```

Comment:
It seems, this line overwrites offset values and push_back in line 120.
I think offset values should be copy from lanelet_pose.

```suggestion
          id, lanelet_pose.s - getLaneletLength(lanelet_pose.lanelet_id), lanelet_pose.offset);
```